pkg/api/types.go

```diff
@@ -1716,6 +1716,40 @@ type NodeStatus struct {
 	NodeInfo NodeSystemInfo `json:"nodeInfo,omitempty"`
 	// List of container images on this node
 	Images []ContainerImage `json:"images",omitempty`
+	// Bounded-sized list of signatures of pods that should avoid this node, sorted
+	// in timestamp order from oldest to newest. Size of the slice is unspecified.
+	PreferAvoidPods []PreferAvoidPodsEntry `json:"preferAvoidPods,omitempty"`
+}
+
+// Describes prefer/avoid rules when scheduling pods to this node.
+type PreferAvoidPodsEntry struct {
+	//  The class of pods.
+	PodSignature PodSignature `json:"podSignature"`
+	// The time at which this eviction happened.
+	EvictionTime unversioned.Time `json:"evictionTime,omitempty"`
+	// (brief) reason for this eviction.
+	Reason string `json:"reason,omitempty"`
+	// Human readable message indicating details about this eviction.
```

Comment:
To make this more general, say "Human-readable message indicating why this entry was added to the list."